### PR TITLE
chore: Add README for new proposed release note process.

### PR DESCRIPTION
**Pull request by neilconway**
## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/release-notes/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+* Update the release notes as part of landing the change itself (e.g., in the
+  same PR or as part of a chain of PRs to land a large feature). The author of
+  the PR has the most context about the change being made and any caveats or
+  additional context that users should be informed about. Reviewers should also
+  look at proposed release note changes as part of reviewing a PR.
+
+* Write the release note entry as a separate file and add it to the
+  `docs/release-notes` directory. Name the file with the PR number as a
+  prefix. For example, `1097-nvidia-a100-support.txt`. The file should be in
```
Comment (hamidzr) [inline docs/release-notes/README.md:24]: :+1:

**Comment by hamidzr** [comment]
since github ui can render `rst` format. it might be also useful if these release notes were part the github PR template. some differences:

- you can see how GitHub would render them easier
- easier to tell what the PR is doing from POV of external "users".
- easier to edit through the UI. harder/no-way to edit from your IDE
- no version control
- it changes how we pull the notes to stitch them together at release time

**Comment by neilconway** [comment]
@hamidzr Yep, that would be another option. Why don't we try the proposed approach and see how it goes? We can discuss in eng-process whether folks want to try a different format/etc.

**Comment by hamidzr** [comment]
> @hamidzr Yep, that would be another option. Why don't we try the proposed approach and see how it goes? We can discuss in eng-process whether folks want to try a different format/etc.

sounds good to me.

File: docs/release-notes/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Release Notes
+
+We want release notes that are accurate, comprehensive, and written so that
```
Comment (hamidzr) [inline docs/release-notes/README.md:3]: missing a "to"

Comment (neilconway) [reply docs/release-notes/README.md:3]: 🤔 hmm, where did you want the "to" added?

**Pull request merged**